interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java

```diff
@@ -463,9 +454,13 @@ public void configure(ClientConfigureRequest request,
             metadata.getValue());
         newMetadata.put(metadata.getType(), md);
       }
-      rpcConfig = new RpcConfig(request.getTypesList(), newMetadata);
-      responseObserver.onNext(ClientConfigureResponse.getDefaultInstance());
-      responseObserver.onCompleted();
+      List<RpcConfig> configs = new ArrayList<>();
+      for (RpcType type : request.getTypesList()) {
+        Metadata md = newMetadata.containsKey(type) ? newMetadata.get(type) : new Metadata();
+        int timeout = request.getTimeoutSec() != 0 ? request.getTimeoutSec() : rpcTimeoutSec;
+        configs.add(new RpcConfig(type, md, timeout));
+      }
+      rpcConfigs = Collections.unmodifiableList(configs);
```

Comment:
`responseObserver` unused?

Reply:
Oops, that's a stupid mistake. 😢  Fixed.